torch/csrc/jit/runtime/vararg_functions.cpp

```diff
@@ -39,6 +39,130 @@ void format(Stack& stack, size_t num_inputs) {
   push(stack, ss.str());
 }
 
+// IValue tags are intentionally private, so we need additional logic to cast
+// the IValue type to the specified format.
+void addFormattedArg(
+    char key,
+    const IValue& ival,
+    std::stringstream& ss,
+    int precision = 6) {
+  // TODO: Implement precison-based formatting
+  switch (key) {
+    case 'd':
+    case 'i': {
+      TORCH_CHECK(
+          ival.isScalar(),
+          "Got ",
+          ival.tagKind(),
+          ", but a number is required for formatting");
+      if (ival.isInt()) {
+        ss << ival.toInt();
+      } else {
+        ss << static_cast<int>(ival.toDouble());
+      }
+      break;
+    }
+    case 'e':
+    case 'E': {
+      TORCH_CHECK(
+          ival.isScalar(),
+          "Got ",
+          ival.tagKind(),
+          ", but a number is required for formatting");
+      ss << std::setprecision(precision) << std::scientific;
+      if (key == 'E') {
+        ss << std::uppercase;
+      }
+      if (ival.isInt()) {
+        ss << static_cast<float>(ival.toInt());
+      } else {
+        ss << static_cast<float>(ival.toDouble());
+      }
+      break;
+    }
+    case 'f':
+    case 'F': {
+      TORCH_CHECK(
+          ival.isScalar(),
+          "Got ",
+          ival.tagKind(),
+          ", but a number is required for formatting");
+      ss << std::setprecision(precision) << std::fixed;
+      if (ival.isInt()) {
+        ss << static_cast<float>(ival.toInt());
+      } else {
+        ss << static_cast<float>(ival.toDouble());
+      }
+      break;
+    }
+    case 'c': {
+      TORCH_CHECK(
+          ival.isInt() || (ival.isString() && ival.toStringRef().length() == 1),
+          "Got ",
+          ival.tagKind(),
+          ", but an int or char is required for formatting");
+      if (ival.isInt()) {
+        ss << static_cast<char>(ival.toInt());
+      } else {
+        ss << ival.toStringRef();
+      }
+      break;
+    }
+    case 's': {
+      if (ival.isString()) {
+        ss << ival.toStringRef();
+      } else {
+        ss << ival;
+      }
+      break;
+    }
+    default: {
+      TORCH_CHECK(
```

Comment:
This error is not tested in your testcases.

Reply:
Also I'd recommend adding a percent sign before key so that it is clearly about formatting.